File: Témata/04 nelineární regresní model (M-H algoritmus)/Matlab/log_post_CES.m
function res = log_post_CES(y,X,gamma,h,gamma_0,V_0)
%res ... logaritmus jadra posteriorni hustoty
%CES funkce
fx = gamma(1)*(gamma(2)*X(:,2).^gamma(4) +gamma(3)*X(:,3).^gamma(4)).^(1/gamma(4));
%logaritmus (5.24) z Koop
res = -1/2*h*(y-fx)'*(y-fx) -1/2*(gamma-gamma_0)'*inv(V_0)*(gamma-gamma_0);
%osetreni nulove hustoty pro parametry gamma
%=> restrikce na kladnost vesch parametru gamma
if min(gamma)<0
    res = -inf;
end

end

File: support/log_post_ces.py
# Funkce pro ziskani logaritmu jadra posteriorni hustoty

import numpy as np
from numpy.linalg import inv

def log_post_ces(y, x, gamma_ast, h, gamma_0, v_0):
    if np.min(gamma_ast) < 0:
        return -np.inf

    # CES funkce
    f_x = gamma_ast[0][0] * \
        ( gamma_ast[1][0] * x[:,[1]] ** gamma_ast[3][0] + gamma_ast[2][0] * x[:,[2]]**gamma_ast[3][0] ) ** \
        ( 1 / gamma_ast[3][0] )
    # Logaritmus (5.24) z Koop
    res = - h / 2 * (y - f_x).T @ (y - f_x) - 1 / 2 * (gamma_ast - gamma_0).T @ inv(v_0) @ (gamma_ast - gamma_0)
    return res
